tests/test_cli.py

```diff
@@ -91,6 +91,12 @@ def setUpClass(cls):
             # dos line endings yaml
             'dos.yml': '---\r\n'
                        'dos: true',
+            # UTF-16 Little Endian BOM
+            'non-ascii/utf16le': b'\xff\xfe---\nutf16le: true\n',
+            # UTF-16 Big Endian
+            'non-ascii/utf16be': b'\xfe\xff---\nutf16be: true\n',
```

Comment:
This test data is wrong.
```py
>>> b'\xff\xfe---\nutf16le: true\n'.decode('utf-16-le')
u'\ufeff\u2d2d\u0a2d\u7475\u3166\u6c36\u3a65\u7420\u7572\u0a65'
```
Better to have u'...'.encode('utf-16-be') - and also need a test that after decoding the files do contain the expected data?

Reply:
Ah, of course. My mistake. UTF-16 and ascii are not compatible, hence the garbage. This would only work correctly for UTF-8. I'll adjust as suggested.

Reply:
I don't quite get why this is wrong. Anyway, does it still need to be adjusted?

Reply:
It's wrong because utf-16 is a two-byte encoding, but the sample data is using `'-'` for instance, not `'u\x00'`.

Reply:
Another illustration of the issue:

```python
In [1]: data = b'\xff\xfe---\nutf16le: true\n'

In [2]: data.decode('utf-16-le')
Out[2]: '\ufeffⴭਭ瑵ㅦ氶㩥琠畲\u0a65'
```

This is because the bytestring uses ascii, which doesn't map back to utf-16.

The correct way looks more like this:

```python
In [4]: b'\xff\xfe' + '---\nutf16le: true'.encode('utf-16-le')
Out[4]: b'\xff\xfe-\x00-\x00-\x00\n\x00u\x00t\x00f\x001\x006\x00l\x00e\x00:\x00 \x00t\x00r\x00u\x00e\x00'

In [5]: _.decode('utf-16-le')
Out[5]: '\ufeff---\nutf16le: true'
```